src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,315 @@
+            var namespaceDeclaration = compilationUnit.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+
+            // Move everything to global namespace
+            if (targetNamespaceParts.Length == 1 && targetNamespaceParts[0].Length == 0)
+            {
+                var (namespaceOpeningTrivia, namespaceClosingTrivia) = 
+                    GetOpeningAndClosingTriviaOfNamespaceDeclaration(namespaceDeclaration);
+                var members = namespaceDeclaration.Members;
+                var eofToken = compilationUnit.EndOfFileToken
+                    .WithAdditionalAnnotations(WarningAnnotation);
+
+                // Try to preserve trivia from original namesapce declaration.
+                // If there's any member inside the declaration, we attach them to the 
+                // first and last member, otherwise, simply attach all to the EOF token.
+                if (members.Count > 0)
+                {
+                    var first = members.First();
+                    var firstWithTrivia = first.WithPrependedLeadingTrivia(namespaceOpeningTrivia);
+                    members = members.Replace(first, firstWithTrivia);
+
+                    var last = members.Last();
+                    var lastWithTrivia = last.WithAppendedTrailingTrivia(namespaceClosingTrivia);
+                    members = members.Replace(last, lastWithTrivia);
+                }
+                else
+                {
+                    eofToken = eofToken.WithPrependedLeadingTrivia(
+                        namespaceOpeningTrivia.Concat(namespaceClosingTrivia));
+                }
+
+                // Moving inner imports out of the namespace declaration can lead to a break in semantics.
+                // For example:
+                //
+                //  namespace A.B.C
+                //  {
+                //    using D.E.F;
+                //  }
+                //
+                //  The using of D.E.F is looked up iwith in the context of A.B.C first. If it's moved outside,
+                //  it may fail to resolve.
+
+                return compilationUnit.Update(
+                    compilationUnit.Externs.AddRange(namespaceDeclaration.Externs), 
+                    compilationUnit.Usings.AddRange(namespaceDeclaration.Usings), 
+                    compilationUnit.AttributeLists, 
+                    members,
+                    eofToken).WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // Change namespace name
+            return compilationUnit.ReplaceNode(namespaceDeclaration, 
+                namespaceDeclaration.WithName(
+                    CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithTriviaFrom(namespaceDeclaration.Name)
+                    .WithAdditionalAnnotations(WarningAnnotation, Formatter.Annotation)));
```

Comment:
nit: all the codepaths attach the Formatter.Annotation at the end.  A cleaner way would be to have a top level function that calls into this one.  This function doesn't add the annotation, but the caller will add it.